components/newsletter-signup/SignupForm.jsx

```diff
@@ -74,7 +77,7 @@ var SignupForm = React.createClass({
     var privacy_policy_link = <a href="https://www.mozilla.org/privacy/websites/"><FormattedMessage id="privacy_policy" defaultMessage="privacy policy"/></a>;
 
     return (
-      <form className="mailinglist-signup" action={process.env.NEWSLETTER_MAILINGLIST_URL} method="POST" onSubmit={this.handleSubmit}>
+      <form className="mailinglist-signup" action={process.env.NEWSLETTER_MAILINGLIST_URL} method="POST">
```

Comment:
`onSubmit` gets triggered only if form has passed browser validation. Since we are not relying on that for form validation, I've moved the event handler to line 96 (as submit button's onClick event handler)